app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt

```diff
@@ -193,41 +234,38 @@ fun ParkingDetailsScreen(
                   modifier = Modifier.fillMaxWidth().testTag("ImagesRow"),
                   horizontalArrangement = Arrangement.spacedBy(8.dp),
                   verticalAlignment = Alignment.CenterVertically) {
+                    // No images
                     if (selectedParking.images.isEmpty()) {
                       Text(
                           text = stringResource(R.string.card_screen_no_image),
                           color = MaterialTheme.colorScheme.onSurface,
                           style = MaterialTheme.typography.bodyMedium,
+                          modifier = Modifier.weight(3f),
+                          textAlign = TextAlign.Left,
                           testTag = "NoImageText")
-                      IconButton(
-                          icon = Icons.Outlined.AddAPhoto,
-                          contentDescription = "Add Image",
-                          onClick = {
-                            Toast.makeText(
-                                    context,
-                                    "A feature to add images will be added later",
-                                    Toast.LENGTH_LONG)
-                                .show()
-                          },
-                          enabled = userSignedIn.value,
-                          testTag = "AddImageIconButton",
-                          modifier = Modifier.padding(start = 8.dp).height(32.dp).fillMaxWidth())
+                      // There are images to display
                     } else {
                       LazyRow(
-                          modifier = Modifier.fillMaxWidth().testTag("ParkingImagesRow"),
+                          modifier = Modifier.weight(2f).testTag("ParkingImagesRow"),
                           horizontalArrangement = Arrangement.spacedBy(8.dp)) {
-                            items(selectedParking.images.size) { index ->
-                              AsyncImage(
-                                  model = selectedParking.images[index],
-                                  contentDescription = "Image $index",
-                                  modifier =
-                                      Modifier.size(170.dp)
-                                          .padding(4.dp)
-                                          .testTag("ParkingImage$index"),
-                                  contentScale = ContentScale.Crop)
+                            items(imagesUrls) { url ->
+                              Image(
+                                  painter = rememberAsyncImagePainter(url),
+                                  contentDescription =
+                                      stringResource(
+                                          R.string.view_profile_screen_profile_picture), // FIXME
+                                  modifier = Modifier.size(100.dp),
```

Comment:
I think here maybe instead of setting the size of the image, we could rather have a fixed height for the lazyrow and have the images fill max height. Honesty not a concern for this PR, as I think the UI could use a bit more work (e.g. when there are images, the add button looks a bit odd hanging there alone). Just taking the chance to mention it and confirm that you agree with me (or not)

Reply:
Sure I'll add this to the list of things to be done